File: 28. Find the Index of the First Occurrence in a String/findtheindexofthefirstoccurrenceinastring.py
class Solution:
    def strStr(self, haystack: str, needle: str) -> int:
        needle_length, haystack_length = len(needle), len(haystack)

        for i in range(haystack_length - needle_length + 1):
            if haystack[i:i + needle_length] == needle:
                return i

        return -1

File: 28. Find the Index of the First Occurrence in a String/findtheindexofthefirstoccurrenceinastring.java
class Solution {
    public int strStr(String haystack, String needle) {
        int needleLength = needle.length(), haystackLength = haystack.length();

        for (int i = 0; i <= haystackLength - needleLength; i++) {
            int j = 0;

            while (j < needleLength && haystack.charAt(i + j) == needle.charAt(j)) {
                j++;
            }

            if (j == needleLength) {
                return i;
            }
        }

        return -1;
    }
}
